tor/src/main/scala/org/bitcoins/tor/client/TorClient.scala

```diff
@@ -0,0 +1,230 @@
+    case None =>
+      val control = InetSocketAddress.createUnresolved(
+        "127.0.0.1",
+        TorParams.DefaultControlPort)
+
+      val auth = SafeCookie()
+      val privKeyPath = conf.datadir.resolve("tor_priv_key")
+
+      TorParams(control, auth, privKeyPath)
+  }
+
+  private lazy val authenticationArg = torParams.authentication match {
+    case Password(_) =>
+      "error"
+//      s"--HashedControlPassword $password" // todo: need to hash the password correctly
+    case _: SafeCookie =>
+      "--CookieAuthentication 1"
+  }
+
+  /** The command to start the daemon on the underlying OS */
+  lazy val cmd: String = {
+
+    val args = Vector(
+      "--ExitRelay 0", // ensure we aren't an exit relay
+      s"--SOCKSPort ${socks5ProxyParams.address.getHostName}:${socks5ProxyParams.address.getPort}",
+      s"--ControlPort ${torParams.controlAddress.getPort}",
+      authenticationArg,
+      s"""--DataDirectory "${conf.torDir.toAbsolutePath}" """,
+      s"""--Log "notice file ${conf.torLogFile.toAbsolutePath}" """
+    ).mkString(" ")
+
+    val executable = TorClient.DEFAULT_TOR_LOCATION match {
+      case Some(default) => default
+      case None          => TorClient.torBinaryFromResource(conf.torDir)
+    }
+
+    s"$executable $args"
+  }
+
+  def isAlive: Boolean = {
+    processOpt match {
+      case Some(p) =>
+        p.isAlive()
+      case None =>
+        false
+    }
+  }
+
+  /** Starts the binary by spinning up a new process */
+  def startBinary(): Future[Unit] = Future {
+    processOpt match {
+      case Some(_) =>
+        //don't do anything as it is already started
+        logger.info(s"Tor binary was already started!")
+        ()
+      case None =>
+        if (cmd.nonEmpty) {
+          val started = process.run()
+          processOpt = Some(started)
+        } else {
+          logger.warn("cmd not set, no Tor binary started")
+        }
+        ()
+    }
+  }
+
+  /** Stops the binary by destroying the underlying operating system process
+    *
+    * If the client is a remote client (not started on the host operating system)
+    * this method is a no-op
+    */
+  def stopBinary(): Future[Unit] = Future {
+    processOpt match {
+      case Some(process) =>
+        if (process.isAlive()) {
+          val _ = process.destroy()
+        }
+        processOpt = None
+      case None =>
+        logger.info(s"No process found, binary wasn't started!")
+        //no process running, nothing to do
+        ()
+    }
+  }
+}
+
+object TorClient extends Logging {
+
+  lazy val DEFAULT_TOR_LOCATION: Option[File] = {
+    def findExecutableOnPath(name: String): Option[File] =
+      sys.env
+        .getOrElse("PATH", "")
+        .split(File.pathSeparator)
+        .map(directory => new File(directory, name))
+        .find(file => file.isFile && file.canExecute)
+
+    if (EnvUtil.isWindows) {
+      findExecutableOnPath("tor.exe")
+    } else {
+      findExecutableOnPath("tor")
```

Comment:
when testing, if you want to get the binaries from bitcoin-s instead of running your local version you should change this value to something like `tor1` so it doesn't find it on the class path